content/departments/product-engineering/engineering/enablement/dev-experience/newsletter.md

```diff
@@ -8,6 +8,123 @@ To learn more about components of Sourcegraph's developer experience, check out
+
+#### Build traces are now uploaded to Honeycomb
+
+Build traces are now uploaded to Honeycomb to dive into the performance of each command that gets run in a pipeline! To learn more, check out the [Pipeline command tracing docs](https://docs.sourcegraph.com/dev/background-information/continuous_integration#pipeline-command-tracing).
+
+#### Test analytics
+
+We have started rolling out Buildkite test analytics support for Go tests and a subset of frontend tests that get run in continuous integration. This is still an experimental Buildkite feature, but you can learn more about it in our [Test analytics docs](https://docs.sourcegraph.com/dev/background-information/continuous_integration#test-analytics).
+
+#### Pipeline documentation
+
+A new command, `sg ci docs`, can now render a full, up-to-date reference of various run types that our pipeline can generate as well as example pipelines of each, such as what gets run with various diff types.
+You can also see a web version of this in the [Pipeline types reference](https://docs.sourcegraph.com/dev/background-information/ci/reference).
+
+Our [pipeline development guide](https://docs.sourcegraph.com/dev/background-information/continuous_integration#pipeline-development) has also been refereshed with updated content, featuring a series of embedded search notebooks! This includes new guidance on:
+
+- [Creating pipeline annotations](https://docs.sourcegraph.com/dev/background-information/continuous_integration#creating-annotations) (using a new API introduced in [#30951](https://github.com/sourcegraph/sourcegraph/pull/30951))
+- [Caching build artefacts](https://docs.sourcegraph.com/dev/how-to/cache_ci_artefacts)
+- [Pipeline observability features](https://docs.sourcegraph.com/dev/background-information/continuous_integration#observability)
+
+#### Generate builds using run types
+
+`sg ci build` now supports an additional argument to automatically generate a Buildkite build using a specified run type ([#30932](https://github.com/sourcegraph/sourcegraph/pull/30932)). For example, to create a `main` dry run build:
+
+```sh
+sg ci build main-dry-run
+```
+
+This now also supports run types that require arguments, such as `docker-images-patch` - learn more in [#31193](https://github.com/sourcegraph/sourcegraph/pull/31193).
+
+![sg ci build](https://user-images.githubusercontent.com/23356519/153933687-110f3657-543b-4d95-b01b-21b6b9e28514.png)
+
+#### Coming soon: stateless Buildkite agents
+
+We will soon be rolling out stateless Buildkite agents to all pipeline builds.
+These should improve the stability and reliability of all pipelines by removing any issues that might be caused by lingering state from other builds.
+Learn more about this in [this Loom demo](https://www.loom.com/share/601c226a8a93429890c40213922476f9)! ([#31003](https://github.com/sourcegraph/sourcegraph/issues/31003))
+
+### Internal tools and libraries
+
+#### Database migrations update
+
+TODO ask eric
```

Comment:
cc @efritz is there anything you want to include here? feel free to just push to the branch or comment with what you'd like to add 😁

Reply:
```suggestion
We have now eradicated two classes of errors related to database migrations.

1. On the site-administrator and ops side, we no longer spuriously mark the database as dirty and give up any attempt at migrations at the first sign of trouble. We no longer immediately fail an upgrade because of the mere presence of an empty table or a concurrently created index. Now we only fail for **actual** reasons.
2. On the development side, we no longer have to worry about two independently created migrations clashing only after both are merged into `main`. That was very annoying to me and now it will never, ever happen again. Check out the help page for the new `sg migration` to check out the new tooling.

See [migrator docs](https://docs.sourcegraph.com/admin/how-to/manual_database_migrations) for additional info.
```

Reply:
Fantastic, thanks!